src/acl_thread.cpp

```diff
@@ -102,6 +103,12 @@ __attribute__((constructor)) static void l_global_lock_init() {
 }
 
 __attribute__((destructor)) static void l_global_lock_uninit() {
+  {
+    std::scoped_lock lock{acl_mutex_wrapper};
+    acl_get_platform()->initialized = 0;
+    acl_signal_condvar(&l_acl_global_condvar); // wake up waiting thread
+  }
+  acl_thread_join(&acl_get_platform()->device_op_queue_update_thread);
```

Comment:
Not sure if this section of code is necessary here? The situation I have in mind is when there is an interrupt coming and waking up the device op queue update thread, and the program is about to unload the library, then the thread will acquire the lock and get preempted, when `acl_reset_condvar` is called it will see a locked lock, and that might result in some problem. But I guess having an interrupt coming in when the program is about to end is pretty rare?

Reply:
Based on the unit tests it seems like this section might still be required, even with this joining code some of the CI jobs are failing when exiting the program:
```
6: TEST(auto_configure, many_ok_forward_compatibility) - 22 ms
6: TEST(auto_configure, simple) - 0 ms
6: 
6: OK (224 tests, 224 ran, 138598587 checks, 0 ignored, 0 filtered out, 14787 ms)
6: 
6: acl_test: ../lib/acl_threadsupport/src/acl_threadsupport.c:372: acl_release_condvar: Assertion `ret == 0' failed.
6/6 Test #6: acl_test .........................Child aborted***Exception:  21.01 sec
```